helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloLogService.java

```diff
@@ -0,0 +1,117 @@
+
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.LogStream;
+import com.spotify.helios.common.descriptors.HostStatus;
+import com.spotify.helios.common.descriptors.JobId;
+import com.spotify.helios.common.descriptors.TaskStatus;
+
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+class HeliosSoloLogService extends AbstractScheduledService {
+
+  private static final Logger log = LoggerFactory.getLogger(HeliosSoloLogService.class);
+
+  private final HeliosSoloDeployment solo;
+  private final DockerClient dockerClient;
+  private final LogStreamProvider logStreamProvider;
+
+  private Map<String, Future> logFutures = Maps.newHashMap();
+
+  HeliosSoloLogService(@NotNull final HeliosSoloDeployment solo,
+                       @NotNull final DockerClient dockerClient,
+                       @NotNull final LogStreamProvider logStreamProvider) {
+    this.solo = solo;
+    this.dockerClient = dockerClient;
+    this.logStreamProvider = logStreamProvider;
+  }
+
+  @Override
+  protected void runOneIteration() throws Exception {
+    try {
+      // fetch all the jobs running on the solo deployment
+      final HostStatus hostStatus = solo.client().hostStatus(solo.agentName()).get();
+      final Map<JobId, TaskStatus> statuses = hostStatus.getStatuses();
+
+      for (final TaskStatus status : statuses.values()) {
+        final JobId jobId = status.getJob().getId();
+        final String containerId = status.getContainerId();
+        if (isNullOrEmpty(containerId)) {
+          continue;
+        }
+
+        if (!logFutures.containsKey(containerId)) {
+          // for any containers we're not already tracking, attach to their stdout/stderr
+          final ScheduledFuture<?> future = this.executor().schedule(new Runnable() {
+            @Override
+            public void run() {
+              try {
+                final LogStream logStream = dockerClient.logs(containerId, stdout(), stderr(),
+                                                              follow());
+
+                log.info("attaching stdout/stderr for job={}, container={}", jobId, containerId);
+                logStream.attach(logStreamProvider.getStdoutStream(jobId, containerId),
+                                 logStreamProvider.getStderrStream(jobId, containerId));
+              } catch (final InterruptedException e) {
+                // we're shutting down, no need to log anything
+              } catch (final Exception e) {
+                log.warn("error streaming log for job={}, container={} - {}", jobId, containerId,
+                         e);
+              }
+            }
+          }, 0, TimeUnit.MILLISECONDS);
```

Comment:
nitpicky but isn't `schedule(Runnable, long, Timeunit)` the same as `submit(Runnable)`?